File: old_fastify/view/team.js
import prisma from '../prisma/script.js';
async function team_add(request, reply) {
    const { name, phone, note } = request.body;
    const add = await prisma.team.create({
        data: {
            name: name,
            phone: phone,
            note: note,
            using: true
        }
    });
    return { status: 0, msg: "成功", data: add };
}
async function team_show(request, reply) {
    let { name, page, pagesize } = request.query;
    // 构建查询条件
    const where = { using: true };
    if (name !== "") {
        where.name = { contains: name };
    }
    // 获取总数
    const total = await prisma.team.count({ where });
    // 获取分页数据
    const show = await prisma.team.findMany({
        where,
        skip: (page - 1) * pagesize,
        take: pagesize,
        orderBy: { id: 'desc' }
    });
    return {
        status: 0,
        msg: "成功",
        data: show,
        total: total, // 使用 count 查询的结果
        page: page,
        pagesize: pagesize,
        totalpages: Math.ceil(total / pagesize) // 使用总数计算总页数
    };
}
async function team_update(request, reply) {
    const { id, name, phone, note, using } = request.body;
    const update = await prisma.team.update({
        where: { id },
        data: {
            name: name,
            phone: phone,
            note: note,
            using: using
        }
    });
    return { status: 0, msg: "成功", data: update };
}
async function team_del(request, reply) {
    const { id } = request.body;
    const del = await prisma.team.update({
        where: {
            id: id
        },
        data: {
            using: false
        }
    });
    return { status: 0, msg: "成功", data: del };
}
export { team_add, team_show, team_update, team_del };

File: LabMangerAPI/Repository/Team.cs

using LabMangerAPI.DTOs;
using LabMangerAPI.Data;
using LabMangerAPI.Models;
using SqlSugar;

namespace LabMangerAPI.Repository;
/// <summary>
/// 小组类（仓储层）
/// </summary>
public class RepositoryTeam:ICrud<Team,Team,RequestTeam.Add,RequestTeam.Show,RequestTeam.Update,RequestTeam.Del>
{
    readonly SqlSugarClient _db=MySqlSugar.Db;
    readonly IUserContext _userContext;
    /// <summary>
    /// 构建小组类(仓储层)
    /// </summary>
    /// <param name="userContext">用户和权限验证接口</param>
    public RepositoryTeam(IUserContext userContext)
    {   
        _userContext = userContext;
    }

    /// <summary>
    /// 增加一个小组
    /// </summary>
    /// <param name="body">增加小组时请求的数据</param>
    ///  <returns>增加一个小组用时返回的数据</returns>
    public async Task<Team> Add(RequestTeam.Add body)
    {
       var result=_db.Insertable(new  Team {
           Name= body.Name,
           Note= body.Note,
           Phone = body.Phone,
       }).ExecuteReturnEntityAsync();
       return await result;
    }
    /// <summary>
    /// 展示多个小组
    /// </summary>
    /// <param name="search">展示小组时请求的数据</param>
    /// <param name="totalcount">ref 小组总数</param>
    /// <param name="totalcount">ref 总页数</param>
    ///  <returns>展示多个小组时返回的数据</returns>
    public async Task<List<Team>> Show(RequestTeam.Show search, RefAsync<int> totalcount, RefAsync<int> totalpage)
    {
        string? name = search.Name;
        int page = search.Page;
        int pageSize = search.PageSize;
        var exp = Expressionable.Create<Team>();
        exp.And(t => t.Status == Status.Enable);
        exp.AndIF(string.IsNullOrEmpty(name) == false, t => t.Name.Contains(name!));
        var result = _db.Queryable<Team>()
            .Where(exp.ToExpression())
            .OrderBy(t => t.Id, OrderByType.Desc)
            .ToPageListAsync(page, pageSize, totalcount, totalpage);

        return await result;

    }
    /// <summary>
    /// 更新一个小组的信息
    /// </summary>
    /// <param name="body">更新小组信息时请求的数据</param>
    ///  <returns>更新小组信息时返回的数据</returns>
    public async Task<Team> Update(RequestTeam.Update body)
    {
        var result = _db.Updateable(new Team {
            Id= body.Id, 
            Name= body.Name,
            Note= body.Note,
            Phone = body.Phone,
        }).ExecuteReturnEntityAsync();
        return await result;
        
    }
    /// <summary>
    /// 删除一个小组的信息
    /// </summary>
    /// <param name="body">删除小组信息时请求的数据</param>
    ///  <returns>删除小组信息时返回的数据</returns>
    public async Task<bool> Del(RequestTeam.Del body)
    {
        await _db.Updateable<Team>()
            .SetColumns(it => it.Status == Status.Delete)
            .Where(it => it.Id == body.Id)
            .ExecuteReturnEntityAsync();
        return  true;
    }
}